api/v1alpha1/hostedcluster_types.go

```diff
@@ -267,6 +273,57 @@ type ClusterAutoscaling struct {
 	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
 }
 
+// EtcdManagementType is a enum specifying the strategy for managing the clusters etcd instance
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type EtcdManagementType string
+
+const (
+	Managed   EtcdManagementType = "Managed"
+	Unmanaged EtcdManagementType = "Unmanaged"
+)
+
+type EtcdSpec struct {
+	// ManagementType defines how the etcd cluster is managed. Unmanaged means
+	// the etcd cluster is managed by a system outside the hypershift controllers.
+	// Managed means the hypershift controllers manage the provisioning of the etcd cluster
+	// and the operations around it
+	// +unionDiscriminator
+	ManagementType EtcdManagementType `json:"managementType"`
+
+	// Managed provides metadata that defines how the hypershift controllers manage the etcd cluster
+	// +optional
+	Managed *ManagedEtcdSpec `json:"managed,omitempty"`
+
+	// Unmanaged provides metadata that enables the Openshift controllers to connect to the external etcd cluster
+	// +optional
+	Unmanaged *UnmanagedEtcdSpec `json:"unmanaged,omitempty"`
+}
+
+type ManagedEtcdSpec struct {
+
+	//TODO: Ultimately backup policies, etc can be defined here.
+}
+
+type UnmanagedEtcdSpec struct {
+	// Endpoint is the full url to connect to the etcd cluster endpoint. An example is
+	// https://etcd-client:2379
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^https://`
+	Endpoint string `json:"endpoint"`
+
+	// TLS defines a reference to a TLS secret that can be used for client MTLS authentication with
+	// the etcd cluster
+	TLS EtcdTLSConfig `json:"tls"`
+}
+
+type EtcdTLSConfig struct {
+	// Client refers to a secret for client MTLS authentication with the etcd cluster
+	// The CA must be stored at secret key etcd-client-ca.crt.
+	// The client cert must be stored at secret key etcd-client.crt.
+	// The client key must be stored at secret key etcd-client.key.
+	Client corev1.LocalObjectReference `json:"client"`
```

Comment:
Would it make sense for this to be named `ClientSecret` to make it clearer that's what it's expected to refer to?

Reply:
I'm fine to name client secret

Reply:
renamed